mobile-app/lib/ui/views/forum/forum-categories/forum_category_viewmodel.dart

```diff
@@ -78,17 +78,21 @@ class ForumCategoryViewModel extends BaseViewModel {
     return prefs.getBool('loggedIn') ?? false;
   }
 
-  Future<User> fetchUser() async {
+  Future<dynamic> fetchUser() async {
     SharedPreferences prefs = await SharedPreferences.getInstance();
-    String username = prefs.getString("username") as String;
 
-    final response = await ForumConnect.connectAndGet('/u/$username');
+    String? username;
 
-    if (response.statusCode == 200) {
-      _userProfileIsLoading = false;
-      notifyListeners();
-      return User.fromJson(jsonDecode(response.body));
+    if (prefs.getString("username") != null) {
+      final response = await ForumConnect.connectAndGet('/u/$username');
```

Comment:
Here `username` will always be null as its initialized but a value is never assigned to it

Reply:
The variable `username` is not checked if it is null. The value from SharedPrefrences is.